types/minimasonry/index.d.ts

```diff
@@ -0,0 +1,79 @@
+// Type definitions for minimasonry 1.3
+// Project: https://spope.github.io/MiniMasonry.js/
+// Definitions by: Paul Ebose <https://github.com/plbstl>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyType
+
+interface MiniMasonryOptions {
+    /**
+     * Target width of elements.
+     * @default 255
+     */
+    baseWidth?: number;
+    /**
+     * Container's selector or element.
+     */
+    container: string | HTMLElement;
+    /**
+     * Width / height of gutter between elements. Use gutterX / gutterY to set different values.
+     * @default 10
+     */
+    gutter?: number;
+    /**
+     * Width of gutter between elements. Need gutterY to work, fallback to `gutter`.
+     */
+    gutterX?: number;
+    /**
+     * Height of gutter between elements. Need gutterX to work, fallback to `gutter`.
+     */
+    gutterY?: number;
+    /**
+     * Whether or not MiniMasonry places elements on the shortest column or keeps exact order of the list.
+     * @default true
+     */
+    minify?: boolean;
+    /**
+     * Set left gutter on first columns and right gutter on last.
+     * @default true
+     */
+    surroundingGutter?: boolean;
+    /**
+     * Gutter applied when only 1 column can be displayed.
+     * @default 5
+     */
+    ultimateGutter?: number;
+    /**
+     * Sorting direction, "ltr" or "rtl".
+     * @default "ltr"
+     */
+    direction?: 'ltr' | 'rtl';
+    /**
+     * False will start to sort from center, true will start from left or right according to direction parameter.
+     * @default false
+     */
+    wedge?: boolean;
+}
+
+/**
+ * You should have a container **relatively positioned** with your elements as children. Those children elements must be **absolutely positioned**.
+ *
+ * MiniMasonry will add a "resize" event listener on the window to redraw the masonry on window resize. This listener is throttled to *66ms (15fps)*.
+ *
+ *  @example
+ * import MiniMasonry from "minimasonry";
+ * const masonry = new MiniMasonry({
+ *     container: '.masonry_transition'
+ * });
+ */
+declare class MiniMasonry {
+    constructor(options: MiniMasonryOptions);
+    /**
+     * If list has changed, trigger a re-layout of the masonry.
+     */
+    layout(): void;
+    /**
+     * Remove the resize listener and set back container as it was before initialization.
+     */
+    destroy(): void;
+}
+
+export = MiniMasonry;
```

Comment:
this looks like ESM module with global script support (as per README), does it support CJS usage?

Reply:
As at right now, no it doesn't support CJS usage

Reply:
As pointed out, this seems to be intended to be used as a browser script, or through a default ESM import, so instead of `export = MiniMasonry` you should use a default export, e.g. `export default class MiniMasonry { ... }`.

And ideally, you should also add types that support the browser usage of this package. In this case, it seems that if you use the package in the browser as a global script, you get a global variable `MiniMasonry`.